### PR TITLE
Add superobbing option to VIIRS ioda converter

### PR DESCRIPTION
**Pull request by ypwang19**
**Description:**

- Update the VIIRS ioda converter with superobbing option.

### CODE REVIEW

File: utils/obsproc/Viirsaod2Ioda.h [hunk 1]
```diff
@@ -132,12 +121,80 @@ namespace gdasapp {
                 }
                 obserror[i][j] = obserrorValue;
                 mask[i][j] = 1;
-                nobs += 1;
              }
           }
         }
       }
 
+      // superobing, not work yet...
```
Comment (CoryMartin-NOAA) [inline utils/obsproc/Viirsaod2Ioda.h]: is this comment correct? or does it work now?

File: utils/obsproc/Viirsaod2Ioda.h [hunk 2]
```diff
@@ -132,12 +121,80 @@ namespace gdasapp {
...
+                lonValue = fmod(lonValue + 360, 360);
+            }
+        }
+
+
+        lat2d_s = gdasapp::superobutils::subsample2D(lat, mask, fullConfig_);
+        mask_s = gdasapp::superobutils::subsample2D(mask, mask, fullConfig_);
+        // implement weighted average
+        if (fullConfig_.has("binning.cressman radius")) {
+        // obsvalue_s = gdasapp::superobutils::subsample2D_cressman(obsvalue, lat, lon,
```
Comment (CoryMartin-NOAA) [inline utils/obsproc/Viirsaod2Ioda.h:167]: maybe a comment here about how this isn't working yet?

File: utils/obsproc/Viirsaod2Ioda.h [hunk 3]
```diff
@@ -81,38 +82,26 @@ namespace gdasapp {
       std::vector<std::vector<float>> obsvalue(dimRow, std::vector<float>(dimCol));
       std::vector<std::vector<float>> obserror(dimRow, std::vector<float>(dimCol));
       std::vector<std::vector<int>> preqc(dimRow, std::vector<int>(dimCol));
-      std::vector<std::vector<float>> seconds(dimRow, std::vector<float>(dimCol));
+//      std::vector<std::vector<float>> seconds(dimRow, std::vector<float>(dimCol));
```
Comment (CoryMartin-NOAA) [inline utils/obsproc/Viirsaod2Ioda.h]: remove if we don't need it

**Review by CoryMartin-NOAA — COMMENTED** [review]
A few minor comments, otherwise looks good, thanks for this!

File: utils/obsproc/Viirsaod2Ioda.h (hunk 1 above)
Comment (ypwang19) [reply utils/obsproc/Viirsaod2Ioda.h]: This comment is removed.

File: utils/obsproc/Viirsaod2Ioda.h (hunk 2 above)
Comment (ypwang19) [reply utils/obsproc/Viirsaod2Ioda.h:167]: comment added.

File: utils/obsproc/Viirsaod2Ioda.h (hunk 3 above)
Comment (ypwang19) [reply utils/obsproc/Viirsaod2Ioda.h]: Removed.

**Comment by CoryMartin-NOAA** [comment]
@ypwang19 coding norm issue but otherwise I'm ready to approve. Thanks for this!

**Review by CoryMartin-NOAA — DISMISSED** [review]
Great work, thanks @ypwang19 !

I'll merge once @andytangborn has a chance to take a look

**Review by andytangborn — APPROVED** [review]
I think it's fine. I was  trying to understand whether the obs errors should just be an average of all the errors in each superobbed set. And whether it might be desirable at some point in the future to use a weighted average of the observations.

**Pull request merged**